src/lib/krad/packet.c

```diff
@@ -222,6 +224,175 @@ packet_set_attrset(krb5_context ctx, const char *secret, krad_packet *pkt)
     return kr_attrset_decode(ctx, &tmp, secret, pkt_auth(pkt), &pkt->attrset);
 }
 
+/* Determines if request or response requires a Message-Authenticator
+ * attribute. */
+inline static krb5_boolean
+requires_msgauth(const char *secret, krad_code code)
+{
+    /* If no secret provided, make the assumption packets exchange will happen
+     * on UNIX socket. Message-Authenticator is required only on UDP and TCP
+     * connections. */
+    if (!secret || 0 == strlen(secret))
```

Comment:
FreeIPA uses `libkrad` to exchange radius messages between 2 local daemons using a UNIX socket. In this case we pass an empty string as secret, since data protection is not needed. I use this fact to determine if verification of `Message-Authenticator` is required or not.

The problem is some applications might be using libkrad on UNIX sockets with a secret. In such case, enforcing `Message-Authenticator` does not make sense.

A better approach would be to check if the message was sent via a UDP or TCP connection, but such information is not available in `krad_packet_decode_request()` nor `krad_packet_decode_response()`.